bootstrap/src/test/java/io/airlift/bootstrap/TestBootstrap.java

```diff
@@ -58,6 +66,42 @@ public void testDoesNotAllowCircularDependencies()
         }
     }
 
+    @Test
+    public void testEnvironmentVariableReplacement()
+    {
+        Map<String, String> original = ImmutableMap.<String, String>builder()
+                .put("apple", "apple-value")
+                .put("grape", "${ENV:GRAPE}")
+                .put("peach", "${ENV:PEACH}")
+                .put("grass", "${ENV:!!!}")
+                .put("pear", "${ENV:X_PEAR}")
+                .put("cherry", "${ENV:X_CHERRY}")
```

Comment:
Add a test case with
```
.put("cherry", "${ENV:X_CHERRY}x")
```

expected value will be `"${ENV:X_CHERRY}x"`, since we replace only whole values, no substring replacement.